sdk/src/command-helper.js

```diff
@@ -43,22 +43,43 @@ class CommandHelper {
    * @param command The command to handle.
    * @private
    */
-  handleCommand(command) {
+  async handleCommand(command) {
+    try {
+      const reply = await this.handleCommandLogic(command);
+      if (reply !== undefined) {
+        this.call.write(reply);
+      }
```

Comment:
But won't it will actually be `{ reply: undefined }` now because of how you changed the undefined check on the reply?

Reply:
This reflects the check previously done here:
https://github.com/lightbend/akkaserverless-javascript-sdk/pull/43/files#diff-87e082fc96a33ca702e912a13787c4a08c23012e221b51cfe101985095b16e8fL190
that comes after the additional wrapping performed in `createReply`, if I understand it correctly it's good like this.

Reply:
You rewrap it in this change though: https://github.com/lightbend/akkaserverless-javascript-sdk/pull/43/files#diff-87e082fc96a33ca702e912a13787c4a08c23012e221b51cfe101985095b16e8fR106 so it won't be the return value that is undefined but a field of it?

Reply:
@johanandren I might actually be wrong here but the wrapping was already done here:
https://github.com/lightbend/akkaserverless-javascript-sdk/pull/43/files#diff-87e082fc96a33ca702e912a13787c4a08c23012e221b51cfe101985095b16e8fL80

```js
reply => { return {reply}; }
```

In general, looking at the code, if we reach that point `reply` is always defined and the check might be redundant.
Should I try to remove the check to validate?

Reply:
Hmm, yea try that and see if things fail.

Reply:
Removed the check, and I'm convinced now that is the right thing to do.
In case of undefined something will throw and it will be caught by the surrounding `try` block.